allennlp/service/predictors/semantic_role_labeler.py

```diff
@@ -61,16 +69,20 @@ def predict_json(self, inputs: JsonDict) -> JsonDict:
         spacy_doc = self.nlp(sentence)
         words = [token.text for token in spacy_doc]
         results = {"words": words, "verbs": []}  # type: JsonDict
-        text = TextField(words, token_indexers=self.token_indexers)
         for i, word in enumerate(spacy_doc):
             if word.pos_ == "VERB":
+                verb = word.text
                 verb_labels = [0 for _ in words]
                 verb_labels[i] = 1
-                verb_indicator = SequenceLabelField(verb_labels, text)
-                output = self.model.tag(text, verb_indicator)
+                instance = self._dataset_reader.text_to_instance(words, verb_labels)
```

Comment:
This hides the viterbi inference logic from the user, which i'm not sure is too great - this change has made it more difficult to use the model in a python script unless you use the `Predictor` (which is of course possible) - I had kind of thought that these classes were meant to be JSON -> JSON wrappers around models with complete functionality. If that's not how you thought of the `Predictor` wrappers, that's fine, but to me, this seems like a blurring of roles.